enumerables.rb

```diff
@@ -0,0 +1,188 @@
+module Enumerable
+  # 1. Create my_each
+  def my_each
+    i = 0
+    while i < length
+      yield self[i] if block_given?
+      i += 1
+    end
+    self unless block_given?
+  end
+
+  # 2. Create my_each_with_index
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i) if block_given?
+      i += 1
+    end
+    self unless block_given?
+  end
+
+  # 3. Create my_select
+  def my_select
```

Comment:
* [ ] should have working my_select  identical and returns the same thing as ruby's[`select`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). 
([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
 
``` p [1,2,3,4,5].my_select`` != ``` p [1,2,3,4,5].select```

``` p [1,2,3,4,5].select``` => should return ```<Enumerator: [1, 2, 3, 4]:select>```

therefore 

```p [1,2,3,4,5]my_select``` => should return``` <Enumerator: [1, 2, 3, 4]:my_select>```
you can handle this by adding this condition to your method

```  return to_enum(the_name_your_method)  unless block_given?```